lua/nvim-treesitter/indent.lua

```diff
@@ -57,6 +56,11 @@ function M.get_indent(lnum)
     return -1
   end
 
+  -- Reparse in case we got triggered by ":h indentkeys"
+  parser:parse()
```

Comment:
while this is probably the right thing to do, it breaks the `join_lines.py` test, which is very pain full for Python devs

Reply:
Reparse also breaks: yaml after foo: | <cr>

Reply:
maybe we, could enable this only for some languages?

Reply:
Doesn't treesitter reparses automatically after each character is inserted in buffer? 🤔

Reply:
No, it doesn't. Currently the reparse executes on every screen redraw: https://github.com/neovim/neovim/blob/7717f38d3f81cc739ef2b37829558d3d9eba9591/runtime/lua/vim/treesitter/highlighter.lua#L301-L305. I've encountered the exactly same problem, the reparse before calculation is definitely needed.

Reply:
Just asking out of curiosity, what is a "buffer redraw" for neovim? If I hit a key on keyboard and that character appears on the buffer, is the buffer redrawn? Or does it refer to somthing else?

Reply:
The redraw refers to the update of the screen display, usually the characters sequence output by neovim to terminal. You can look at https://github.com/neovim/neovim/blob/master/src/nvim/screen.c for more detailed explanation.